frontend/src/components/Simulator/ExecuteContractMethods.vue

```diff
@@ -31,11 +32,16 @@ const inputs = computed<{ [k: string]: any }>(() => {
   }, {})
 })
 
-
 watch(() => props.abi, (newValue) => {
   method.value = undefined
-  methodList.value =  Object.entries(newValue?.methods || {})
-    .filter((m) => !m[0].startsWith('_') && !m[0].startsWith('get_'))
+  methodList.value = Object.entries(newValue?.methods || {})
+    .filter((m) => {
+      if (props.mode === 'read') {
```

Comment:
```suggestion
    const isGetterMethod = m[0].startsWith('get_')
    const isPrivateMethod = m[0].startsWith('_')
    
    return !isPrivateMethod && (props.mode === 'read' ? isGetterMethod : !isGetterMethod)
```

Reply:
@cristiam86 , in this case you mean that is better to use a single component for both and just render the two sections ?

Reply:
Probably yes but I wont revert those changes now that they have been made

Reply:
@cristiam86 fixed now I used the `ExecuteContractMethods` as a container and then Another component that list the methods and we separated by getters and writters 

About the fixes and refactor will take in account for next situation and create a separated PR with fixes before the new feature.